alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/implementations/reactions/AbstractOrientingBehavior.kt

```diff
@@ -202,9 +211,7 @@ abstract class AbstractOrientingBehavior<T, P, A, N, E, M, F>(
         }
         val rankings = if (route.isNotEmpty()) {
             computeEdgeRankings(currRoom, route[0].centroid)
-        } else {
-            null
-        }
+        } else null
```

Comment:
I feel like I've seen this code before. There is duplication: can you extract this "rankings" into a protected method in a common abstract class?

Reply:
(or a protected extension function of Route, whatever is cleaner)